tests/samplers_tests/test_nsgaii.py

```diff
@@ -304,266 +306,115 @@ def test_constrained_dominates_infeasible_vs_infeasible(direction: StudyDirectio
-        _create_frozen_trial(1, [10, 10], [0]),
-        _create_frozen_trial(2, [20, 20], [0]),
-        _create_frozen_trial(3, [30, 10], [0]),
-        _create_frozen_trial(4, [15, 15], [0]),
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {0, 2, 3},
-        {4},
-        {1},
-    ]
-
-    # Three objective.
-    directions = [StudyDirection.MAXIMIZE, StudyDirection.MAXIMIZE, StudyDirection.MINIMIZE]
-    trials = [
-        _create_frozen_trial(0, [5, 5, 4], [0]),
-        _create_frozen_trial(1, [5, 5, 5], [0]),
-        _create_frozen_trial(2, [9, 9, 0], [0]),
-        _create_frozen_trial(3, [5, 7, 5], [0]),
-        _create_frozen_trial(4, [0, 0, 9], [0]),
-        _create_frozen_trial(5, [0, 9, 9], [0]),
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {2},
-        {0, 3, 5},
-        {1},
-        {4},
-    ]
+                    assert not _constrained_dominates(trial1, trial2, direction)
 
+        # Check that each trial is dominated by some trial in the rank above.
+        for i in range(len(population_per_rank) - 1):
+            for trial2 in population_per_rank[i + 1]:
+                assert any(
+                    _constrained_dominates(trial1, trial2, direction)
+                    for trial1 in population_per_rank[i]
+                )
 
-def test_fast_non_dominated_sort_constrained_infeasible() -> None:
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
-        sampler = NSGAIISampler(constraints_func=lambda _: [0])
-
-    # Single objective.
-    directions = [StudyDirection.MINIMIZE]
-    trials = [
-        _create_frozen_trial(0, [10], [1]),
-        _create_frozen_trial(1, [20], [3]),
-        _create_frozen_trial(2, [20], [2]),
-        _create_frozen_trial(3, [30], [1]),
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {0, 3},
-        {2},
-        {1},
-    ]
 
-    # Two objective.
-    directions = [StudyDirection.MAXIMIZE, StudyDirection.MAXIMIZE]
-    trials = [
-        _create_frozen_trial(0, [10, 30], [1, 1]),
-        _create_frozen_trial(1, [10, 10], [2, 2]),
-        _create_frozen_trial(2, [20, 20], [3, 3]),
-        _create_frozen_trial(3, [30, 10], [2, 4]),
-        _create_frozen_trial(4, [15, 15], [4, 4]),
-    ]
-    population_per_rank = sampler._fast_non_dominated_sort(trials, directions)
-    assert [{t.number for t in population} for population in population_per_rank] == [
-        {0},
-        {1},
-        {2, 3},
-        {4},
-    ]
+@pytest.mark.parametrize("values_dim", [1, 2, 3])
```

Comment:
How about splitting each dimension test into separate functions as we done for `_constrained_dominates`?

Reply:
Unlike `_constrained_dominates`, this does not make the test easier, since we need to get the whole list anyways.